clientapi/routing/register.go

```diff
@@ -581,11 +588,33 @@ func handleRegistrationFlow(
 	// TODO: Handle mapping registrationRequest parameters into session parameters
 
 	// TODO: email / msisdn auth types.
+	accessToken, accessTokenErr := auth.ExtractAccessToken(req)
```

Comment:
Can we deal with `accessTokenErr` ASAP please rather than rely on a complicated if statement on `:595` and a subsequent function call? Seems confusing.